Sources/SuperwallKit/Network/Network.swift

```diff
@@ -0,0 +1,198 @@
+//
+//  ModelLoader.swift
+//  Superwall
+//
+//  Created by Yusuf Tör on 04/03/2022.
+//
+
+import Foundation
+import UIKit
+import Combine
+
+class Network {
+  private let urlSession: CustomURLSession
+  private var applicationStatePublisher: AnyPublisher<UIApplication.State, Never> {
+    UIApplication.shared.publisher(for: \.applicationState)
+      .eraseToAnyPublisher()
+  }
+  private let factory: ApiFactory
+
+  init(
+    urlSession: CustomURLSession = CustomURLSession(),
+    factory: ApiFactory
+  ) {
+    self.urlSession = urlSession
+    self.factory = factory
+  }
+
+  func sendEvents(events: EventsRequest) async {
+    do {
+      let result = try await urlSession.request(.events(eventsRequest: events, factory: factory))
+      switch result.status {
+      case .ok:
+        break
+      case .partialSuccess:
+        Logger.debug(
+          logLevel: .warn,
+          scope: .network,
+          message: "Request had partial success: /events",
+          info: ["payload": result.invalidIndexes as Any]
+        )
+      }
+    } catch {
+      Logger.debug(
+        logLevel: .error,
+        scope: .network,
+        message: "Request Failed: /events",
+        info: ["payload": events],
+        error: error
+      )
+    }
+  }
+
+  func getPaywall(
+    withId identifier: String? = nil,
+    fromEvent event: EventData? = nil
+  ) async throws -> Paywall {
+    do {
+      return try await urlSession.request(
+        .paywall(withIdentifier: identifier, fromEvent: event, factory: factory)
+      )
+    } catch {
+      if identifier == nil {
+        Logger.debug(
+          logLevel: .error,
+          scope: .network,
+          message: "Request Failed: /paywall",
+          info: [
+            "identifier": identifier ?? "none",
+            "event": event.debugDescription
+          ],
+          error: error
+        )
+      } else {
+        Logger.debug(
+          logLevel: .error,
+          scope: .network,
+          message: "Request Failed: /paywall/:identifier",
+          error: error
+        )
+      }
+      throw error
+    }
+  }
+
+  func getPaywalls() async throws -> [Paywall] {
+    do {
+      let response = try await urlSession.request(
+        .paywalls(factory: factory),
+        isForDebugging: true
```

Comment:
Do you want `isForDebugging: true` here? I'm also not sure I see it being used anywhere

Reply:
Huh I can't remember why that was there but yeah it's not used

Reply:
Oh wait it was for setting the apiKey when launching the debugger, but there was just an easier way to do it so I removed it from that func